lib/fetching/content-services/facebook-content-service.js

```diff
@@ -26,116 +27,139 @@ FacebookContentService.prototype._doFetch = function(options, callback) {
   var self = this;
 
   if (!this._url) {
-    process.nextTick(function() { self.emit('error', new Error('Missing Facebook URL')); });
+    process.nextTick(function () {
+      self.emit('error', new Error('Missing Facebook URL'));
+    });
     callback([]);
     return;
   }
 
-  var queries = {};
+  var source = this._url.trim();
 
-  // Determine Source ID of object
-  queries.source = 'SELECT id, site, type, url FROM object_url WHERE url = "' + this._url + '"';
+  // Parse out source name from url
+  if (source.indexOf('?') === -1) {
+    source = source.split('?')[0];
+  }
 
-  // Fetch posts and comments
-  queries.posts = 'SELECT source_id, post_id, updated_time, created_time, actor_id, permalink, message';
-  queries.posts += ' FROM stream';
-  queries.posts += ' WHERE source_id IN (SELECT id FROM #source)';
-  // New comments trigger a fresh updated_time in its parent post
-  if (this._lastReportDate) queries.posts += ' AND updated_time > ' + (this._lastReportDate.getTime()/1000);
-  queries.posts += ' ORDER BY updated_time';
+  if (source.lastIndexOf('/') === source.length - 1) {
+    source = source.substring(0, source.length - 1);
+  }
 
-  // Get new comments associated with posts.
-  queries.comments = "SELECT id, fromid, time, text, post_id FROM comment WHERE (post_id IN (SELECT post_id FROM #posts))";
-  if (this._lastReportDate) queries.comments += ' AND time > ' + (this._lastReportDate.getTime()/1000);
-  queries.comments += ' ORDER BY time';
+  source = source.substring(source.lastIndexOf('/') + 1, source.length);
```

Comment:
I would strongly suggest using a regular expression to parse this instead of all these calls to substring, indexOf, etc.